### PR TITLE
Fix border number handling

### PR DESCRIPTION
**Pull request by anne-ds**
Fixes: [RISDEV-5038](https://digitalservicebund.atlassian.net/browse/RISDEV-5038) and [RISDEV-5072](https://digitalservicebund.atlassian.net/browse/RISDEV-5072)

[RISDEV-5038]: https://digitalservicebund.atlassian.net/browse/RISDEV-5038?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RISDEV-5072]: https://digitalservicebund.atlassian.net/browse/RISDEV-5072?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: frontend/src/editor/commands/addBorderNumbers.ts [hunk 1]
```diff
@@ -17,114 +12,55 @@ function addBorderNumbers({ state, dispatch }: CommandProps): boolean {
...
 
-    if (isParagraphWithContent(node, paragraphNodeType)) {
-      const borderNumberNode = createBorderNumberNode(
-        node,
-        schema,
-        contentNodeType,
-      )
-      const currentPos = pos + shift
+    const isBorderNumberNode =
+      node.type === borderNumberNodeType || node.hasMarkup(borderNumberNodeType)
```
Comment (anne-ds) [inline frontend/src/editor/commands/addBorderNumbers.ts:26]: @kiwikern I don't understand why we need node.hasMarkup here?

File: frontend/src/editor/commands/addBorderNumbers.ts [hunk 2]
```diff
@@ -17,114 +12,55 @@ function addBorderNumbers({ state, dispatch }: CommandProps): boolean {
...
-
-      modified = true
-    }
+    nodesWithoutBorderNumbers.push({ node, pos })
   })
 
+  const reversedNodes = [...nodesWithoutBorderNumbers].reverse()
+  for (const { node, pos } of reversedNodes) {
+    const numberNode = numberNodeType.create(
+      null,
```
Comment (anne-ds) [inline frontend/src/editor/commands/addBorderNumbers.ts:41]: Why do we need these null values while creating?

File: frontend/src/editor/commands/addBorderNumbers.ts (hunk 1 above)
Comment (kiwikern) [reply frontend/src/editor/commands/addBorderNumbers.ts:27]: Does not make sense, can be removed. 👍

File: frontend/src/editor/commands/addBorderNumbers.ts (hunk 2 above)
Comment (kiwikern) [reply frontend/src/editor/commands/addBorderNumbers.ts:42]: If I remember correctly, I believe the first param are the attributes for the node. null would mean there are none. We don't have any for new ones, right?

File: frontend/src/editor/commands/addBorderNumbers.ts (hunk 1 above)
Comment (anne-ds) [reply frontend/src/editor/commands/addBorderNumbers.ts:27]: Done

**Pull request merged**